From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dan Muey <dan@cpanel.net>
Date: Wed, 21 Sep 2022 17:57:21 +0000
Subject: [PATCH 4/4] Allow Lua 5.4

---
 apache2/msc_lua.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/apache2/msc_lua.c b/apache2/msc_lua.c
index 51be174..e037ac7 100644
--- a/apache2/msc_lua.c
+++ b/apache2/msc_lua.c
@@ -429,12 +429,12 @@ int lua_execute(msc_script *script, char *param, modsec_rec *msr, msre_rule *rul
 #else
 
     /* Create new state. */
-#if LUA_VERSION_NUM == 502 || LUA_VERSION_NUM == 503 || LUA_VERSION_NUM == 501
+#if LUA_VERSION_NUM == 502 || LUA_VERSION_NUM == 503 || LUA_VERSION_NUM == 501 || LUA_VERSION_NUM == 504
     L = luaL_newstate();
 #elif LUA_VERSION_NUM == 500
     L = lua_open();
 #else
-#error We are only tested under Lua 5.0, 5.1, 5.2, or 5.3.
+#error We are only tested under Lua 5.0, 5.1, 5.2, or 5.3 (5.4).
 #endif
     luaL_openlibs(L);
 
@@ -459,10 +459,10 @@ int lua_execute(msc_script *script, char *param, modsec_rec *msr, msre_rule *rul
     /* Register functions. */
 #if LUA_VERSION_NUM == 500 || LUA_VERSION_NUM == 501
     luaL_register(L, "m", mylib);
-#elif LUA_VERSION_NUM == 502 || LUA_VERSION_NUM == 503
+#elif LUA_VERSION_NUM == 502 || LUA_VERSION_NUM == 503 || LUA_VERSION_NUM == 504
     luaL_setfuncs(L, mylib, 0);
 #else
-#error We are only tested under Lua 5.0, 5.1, 5.2, or 5.3.
+#error We are only tested under Lua 5.0, 5.1, 5.2, or 5.3 (5.4).
 #endif
 
     lua_setglobal(L, "m");
