--- a/apache2/re.c	2016-12-15 23:08:46.021000000 -0600
+++ b/apache2/re.c	2016-12-15 23:10:13.317000000 -0600
@@ -1889,8 +1889,7 @@
             }
         }
         else if (rc < 0) {
-            msr_log(msr, 1, "Rule processing failed.");
-
+            msr_log(msr, 1, "Rule processing failed (id=%s)", rule->actionset->id);
 
             if (msr->txcfg->reqintercept_oe == 1)   {
                 apr_table_clear(msr->matched_vars);
